### ISSUE TITLE
install.sh causes System.TypeInitializationException

### ISSUE THREAD

**Issue by hiroakit** [hiroakit/profile#22]
# Summary

When running `install.sh`, dotnet throws `System.TypeInitializationException` and the process is interrupted.

# Environment

- Apple M1 Max
- macOS Sonoma v14.3
  - Hosted on UTM v4.4.5 (94)
  - RAM 4GB
- Homebrew v4.2.5
- .NET 6.0.2 ([Release Notes](https://github.com/dotnet/core/blob/main/release-notes/6.0/6.0.2/6.0.2.md?WT.mc_id=dotnet-35129-website))
  - SDK 6.0.102
  - `install.sh (scripts/install-dotnet.sh)` use dotnet-sdk-6.0.102-osx-arm64.pkg
    - https://download.visualstudio.microsoft.com/download/pr/cc2a94b1-7f3c-44f8-a842-f288a0cff04e/32446a93655522a1f933d4afb5e15836/dotnet-sdk-6.0.102-osx-arm64.pkg
    - Checksum SHA512 ([6.0.2-sha.txt](https://dotnetcli.blob.core.windows.net/dotnet/checksums/6.0.2-sha.txt)) 
      - e4e2738a87ad097b1e5f6eb7b7526d713ba80a479e545c52ddac8413db026f7bb611e2803d4ae7a49780c7d8479c16533723f4f6b2f7b291312ffcb915388c61
    
# Steps to reproduce

```
git clone https://github.com/hiroakit/profile
cd profile
sh install.sh
```

**Comment by hiroakit** [comment]
I installed .NET SDK 6.0.102 ([dotnet-sdk-6.0.102-osx-arm64.pkg](https://download.visualstudio.microsoft.com/download/pr/cc2a94b1-7f3c-44f8-a842-f288a0cff04e/32446a93655522a1f933d4afb5e15836/dotnet-sdk-6.0.102-osx-arm64.pkg)) by GUI. 
but dotnet command raise exception.

```
$ which dotnet
/usr/local/share/dotnet/dotnet
```

```
$ dotnet --info
Unhandled exception. System.TypeInitializationException: The type initializer for 'Microsoft.DotNet.Cli.Utils.Reporter' threw an exception.
 ---> System.TypeInitializationException: The type initializer for 'System.ConsolePal' threw an exception.
 ---> System.OutOfMemoryException: Array dimensions exceeded supported range.
   at System.ConsolePal..cctor()
   --- End of inner exception stack trace ---
   at System.Console.get_ForegroundColor()
   at Microsoft.DotNet.Cli.Utils.AnsiConsole..ctor(TextWriter writer)
   at Microsoft.DotNet.Cli.Utils.Reporter.Reset()
   at Microsoft.DotNet.Cli.Utils.Reporter..cctor()
   --- End of inner exception stack trace ---
   at Microsoft.DotNet.Cli.Utils.Reporter.get_Error()
   at Microsoft.DotNet.Cli.Program.Main(String[] args)
```

When I try dotnet new version, the exception doesn't occur.

- .NET 6.0.26 ([Release notes](https://github.com/dotnet/core/blob/main/release-notes/6.0/6.0.26/6.0.26.md?WT.mc_id=dotnet-35129-website))
  - .NET SDK 6.0.126
    - [dotnet-sdk-6.0.126-osx-arm64.pkg](https://download.visualstudio.microsoft.com/download/pr/1bfd8adc-359a-4fb0-b2d1-0cad866b4ea3/04cb083ce577a7ba60b8c46b6d25a72a/dotnet-sdk-6.0.126-osx-arm64.pkg)
    - Checksum ([source](https://dotnetcli.blob.core.windows.net/dotnet/checksums/6.0.26-sha.txt))
      - 35733a80b148309f4c7ccc6035b76188ac30f41ea17e364016c4f868a1b6af214234f7fce96114c9ecb3b6b8d2a15216627a86cc5599beb599a0e0cc67f2134a

**Comment by hiroakit** [comment]
Fixed #22
